build-system/tasks/build.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const $$ = require('gulp-load-plugins')();
+const gulp = $$.help(require('gulp'));
+const {
+  buildAlp,
+  buildExaminer,
+  buildWebWorker,
+  compile,
+  compileJs,
+  printConfigHelp,
+  printNobuildHelp,
+} = require('./helpers');
+const {buildExtensions} = require('./extension-helpers');
+const {compileCss} = require('./css');
+const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
+const {isTravisBuild} = require('../travis');
+const {parseExtensionFlags} = require('./extension-helpers');
+
+const maybeUpdatePackages = isTravisBuild() ? [] : ['update-packages'];
+
+/**
+ * Enables watching for file changes in css, extensions.
+ * @return {!Promise}
+ */
+function watch() {
+  const handlerProcess = createCtrlcHandler('watch');
+  return performBuild(true).then(() => exitCtrlcHandler(handlerProcess));
+}
+
+/**
+ * Main build
+ * @return {!Promise}
+ */
+function build() {
+  const handlerProcess = createCtrlcHandler('build');
+  return performBuild().then(() => exitCtrlcHandler(handlerProcess));
+}
+
+/**
+ * Performs the build steps for gulp build and gulp watch
+ * @param {boolean} watch
+ * @return {!Promise}
+ */
+function performBuild(watch) {
```

Comment:
Since these methods are moving, it might be worth converting them to `async` functions, since they already return promises.